packages/alfa-dom/src/matches.ts

```diff
@@ -430,12 +430,24 @@ function matchesPseudoClass(
 
     // https://drafts.csswg.org/css-scoping/#host-selector
     case "host":
+      // Do not allow prefix (e.g. "div:host")
+      if (root !== selector) {
+        return false;
+      }
+
+      if (options.treeContext === undefined) {
+        return false;
+      }
+
+      const host = getParentElement(options.treeContext, context, {
+        composed: true
+      });
+      const isHostMatched =
+        selector.value === null ||
+        matches(element, context, selector.value, options, root);
+
```

Comment:
Doing this...

```ts
if (host !== element) {
  return false;
}

return selector.value === null || matches(element, context, selector.value, options, root);
```

...will ensure that we don't attempt to do selector matching unless we absolutely have to.